src/tls_stream.rs

```diff
@@ -58,13 +60,27 @@ impl Builder {
     /// successful. The Ok() variant indicates a successful validation while the Err() variant  
     /// contains the errorcode returned from the internal verification process.    
     /// The validated certificate, is accessible through the second argument of the closure.
+    #[deprecated(note = "please use verify_callback2 instead")]
     pub fn verify_callback<F>(&mut self, callback: F) -> &mut Builder 
         where F: Fn(io::Result<()>, &CertChain) -> io::Result<()> + 'static
     {
         self.verify_callback = Some(Arc::new(callback));
         self
     }
 
+    /// Set a verification callback to be used for connections created with this `Builder`.
+    ///
+    /// The callback is provided with an io::Result indicating if the (pre)validation was  
+    /// successful. The Ok() variant indicates a successful validation while the Err() variant  
+    /// contains the errorcode returned from the internal verification process.    
+    /// The validated certificate, is accessible through the second argument of the closure.
+    pub fn verify_callback2<F>(&mut self, callback: F) -> &mut Builder 
```

Comment:
I don't think we've cut a release with `verify_callback` so we can just update this in place.